Analysis/src/EventAnalyzerCore.cpp

```diff
@@ -41,6 +41,13 @@ EventAnalyzerCore::EventAnalyzerCore(const CoreAnalyzerArguments& args, Channel
         throw exception("Setup '%1%' not found in the configuration file '%2%'.") % args.setup() % args.sources();
     ana_setup = ana_setup_collection.at(args.setup());
 
+    SignalObjectSelector signalObjectSelector(ana_setup.mode);
```

Comment:
`signalObjectSelector = SignalObjectSelector(ana_setup.mode);` and have it as a private member (see comment above)